pallets/elections/src/impls.rs

```diff
@@ -173,19 +176,34 @@ impl<T> Pallet<T>
         )
     }
 
+    fn if_era_starts_do<F: Fn()>(era: EraIndex, start_index: SessionIndex, on_era_start: F) {
+        if let Some(era_start_index) = pallet_staking::ErasStartSessionIndex::<T>::get(era) {
+            if era_start_index == start_index {
+                on_era_start()
+            }
+        }
+    }
+
     fn populate_reserved_on_next_era_start(start_index: SessionIndex) {
         let current_era = match pallet_staking::ActiveEra::<T>::get() {
             Some(ae) => ae.index,
             _ => return,
         };
         // this will be populated once for the session `n+1` on the start of the session `n` where session
         // `n+1` starts a new era.
-        if let Some(era_index) = pallet_staking::ErasStartSessionIndex::<T>::get(current_era + 1) {
-            if era_index == start_index {
-                let reserved_validators = ReservedMembers::<T>::get();
-                ErasReserved::<T>::put(reserved_validators);
-            }
-        }
+        Self::if_era_starts_do(current_era + 1, start_index, || {
+            let reserved_validators = ReservedMembers::<T>::get();
+            ErasReserved::<T>::put(reserved_validators)
+        });
+    }
+
+    fn populate_totals_on_next_era_start(start_index: SessionIndex) {
```

Comment:
Why `next` in name. It should and works for current era, right? `_on_new_era_start`?